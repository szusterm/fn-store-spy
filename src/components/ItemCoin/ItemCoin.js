import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class ItemCoin extends Component {
	render() {
		const {imageSrc, rarity} = this.props;

		return (
			<div className={`item-coin item-coin--${rarity}`}>
				<img
					className={'item-coin--image'}
					src={imageSrc}
				/>
			</div>
		);
	}
}

ItemCoin.propTypes = {
	_id: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired
};

export default ItemCoin;