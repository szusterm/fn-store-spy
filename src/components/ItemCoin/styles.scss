@import '../../scss/rarities-colors';
@import '../../scss/shadow';

.item-coin {
	width: 100%;
	height: auto;

	border-radius: 50%;
	border: 3px solid;

	@include shadow(sm);

	&--uncommon {
		background: $gradient-uncommon;
		border-color: $border-uncommon;
	}
	&--rare {
		background: $gradient-rare;
		border-color: $border-rare;
	}
	&--epic {
		background: $gradient-epic;
		border-color: $border-epic;
	}
	&--legendary {
		background: $gradient-legendary;
		border-color: $border-legendary;
	}

	.item-coin {
		&--image {
			width: 100%;
			height: 100%;

			border-radius: 50%;
		}
	}
}