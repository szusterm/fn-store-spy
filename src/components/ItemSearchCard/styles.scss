@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '../../scss/variables';
@import '../../scss/shadow';
@import '../../scss/transition';
@import '../../scss/opacity-colors';
@import '../../scss/button-reset';

.item-search-card {
	$padding: 10px;

	@include button-reset();
	cursor: pointer;

	max-width: 100%;
	height: auto;
	padding: $padding;

	&--top-part, &--bot-part {
		border: 2px solid transparent;
		@include white-background(1);
		@include transition(md);

		transition-property: box-shadow, border;
	}

	&:hover {
		.item-search-card {
			&--top-part, &--bot-part {
				.visual-box--action-box {
					opacity: 1;

					&--active {
						.action-box--sign {
							@include media-breakpoint-up(sm) {
								background-color: $color-red;
							}
						}
					}

					.action-box--sign {
						transform: scale(1);
					}
				}
			}
		}
	}

	&:focus {
		.item-search-card {
			&--top-part, &--bot-part {
				border-color: rgba(0, 0, 0, 0.3);
			}
		}
	}

	&--top-part {
		$padding: 12px;

		padding: $padding;

		border-top-left-radius: 50%;
		border-top-right-radius: 50%;
		border-bottom: 0;

		position: relative;

		.top-part--visual-box {
			.visual-box--action-box {
				&--active {
					opacity: 1;

					.action-box--sign {
						transform: scale(1);
						background-color: $color-green;

						&:hover {
							background-color: $color-red;
						}
					}
				}

				position: absolute;
				top: 0;
				bottom: 0;
				left: 0;
				right: 0;

				display: flex;
				justify-content: center;
				align-items: center;

				margin: $padding;
				border-radius: 50%;
				opacity: 0;

				@include black-background(0.7);
				@include transition(sh);
			}
		}

		.action-box {
			&--sign {
				padding: 10px;
				border-radius: 50%;
				color: white;
				font-size: 3em;

				transform: scale(0);

				@include black-background(1);
				@include transition(md);
			}
		}
	}

	&--bot-part {
		$padding: 15px;

		height: 110px;

		border-bottom-left-radius: 10px;
		border-bottom-right-radius: 10px;
		border-top: 0;

		display: flex;
		flex-direction: column;
		justify-content: space-between;
		align-items: center;

		.bot-part {
			text-align: center;

			&--name {
				font-size: 1.1em;
				font-weight: 500;
				text-align: center;
				padding: 0 $padding;
			}

			&--description {
				width: 100%;
				padding: $padding;

				display: flex;
				flex-direction: row;
				justify-content: space-around;
				align-items: center;

				.description {
					&--type {
						font-size: 0.8em;
						font-weight: 400;
					}

					&--price {
						font-size: 1em;
						font-weight: 100;
					}
				}
			}
		}
	}
}