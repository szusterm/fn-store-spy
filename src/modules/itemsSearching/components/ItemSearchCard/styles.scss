@import '~material-colors/dist/colors';

@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins/breakpoints';

@import '~scss/_shadow';
@import '~scss/_transition';
@import '~scss/_opacity-colors';
@import '~scss/_input';
@import '~scss/_center-items';

.item-search-card {
	$strip-height: 16px;
	$padding: 10px;

	@include input-reset();
	cursor: pointer;

	max-width: 100%;
	height: auto;
	padding: $padding;

	&--top-part, &--bot-part {
		@include white-background(1);
		@include transition(md);
		@include shadow(sm);

		transition-property: box-shadow, border;
	}

	&:hover {
		.item-search-card {
			&--top-part, &--bot-part {
				@include shadow(md);

				.visual-box--action-box {
					opacity: 1;

					&--active {
						.action-box--sign {
							@include media-breakpoint-up(sm) {
								background-color: $md-red-500;
							}
						}
					}

					.action-box--sign {
						transform: scale(1);
					}
				}
			}
		}
	}

	&:focus {
		.item-search-card {
			&--top-part, &--bot-part {
				@include shadow(md);
			}
		}
	}

	&--top-part {
		$padding: 12px;

		padding: $padding;

		border-top-left-radius: 50%;
		border-top-right-radius: 50%;
		border-bottom: 0;

		position: relative;

		&::after {
			width: 100%;
			background: $body-bg;
			height: $strip-height;
			content: "";

			display: block;
			position: absolute;
			top: calc(100% - #{$strip-height}/2);
			left: 0;
			right: 0;
		}

		.top-part--visual-box {
			.visual-box--action-box {
				&--active {
					opacity: 1;

					.action-box--sign {
						transform: scale(1);
						background-color: $md-green-500;

						&:hover {
							background-color: $md-red-500;
						}
					}
				}

				position: absolute;
				top: 0;
				bottom: 0;
				left: 0;
				right: 0;

				margin: $padding;
				border-radius: 50%;
				opacity: 0;

				@include black-background(0.7);
				@include transition(sh);
				@include center-items();
			}
		}

		.action-box {
			&--sign {
				padding: 10px;
				border-radius: 50%;
				font-size: 3em;

				transform: scale(0);

				@include white-color(1);
				@include black-background(1);
				@include transition(md);
			}
		}
	}

	&--bot-part {
		$padding: 15px;

		padding-top: $strip-height/2;
		height: 110px;

		border-bottom-left-radius: 10px;
		border-bottom-right-radius: 10px;

		display: flex;
		flex-direction: column;
		justify-content: space-between;
		align-items: center;

		.bot-part {
			text-align: center;

			&--name {
				font-size: 1.1em;
				font-weight: 500;
				text-align: center;
				padding: 0 $padding;
			}

			&--description {
				width: 100%;
				padding: $padding;

				@include center-items();

				.description {
					&--price {
						font-size: 1em;
						font-weight: 100;
					}
				}
			}
		}
	}
}