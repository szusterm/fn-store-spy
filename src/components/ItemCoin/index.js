import ItemCoin from './ItemCoin';

export default ItemCoin;