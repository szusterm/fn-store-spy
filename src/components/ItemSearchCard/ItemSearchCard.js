import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {addItemToOrder, removeItemFromOrderByIndex} from '../../redux/actions';

import ItemCoin from '../ItemCoin';

import './styles.scss';

export class ItemSearchCard extends Component {
	constructor() {
		super();

		this.switchOrder = this.switchOrder.bind(this);
		this.checkThatItemIsOrdered = this.checkThatItemIsOrdered.bind(this);
	}

	switchOrder() {
		const {_id, name, rarity, imageSrc} = this.props;

		const indexInOrder = this.getIndexInOrder();

		if (indexInOrder === null) {
			const item = {
				_id,
				name,
				rarity,
				imageSrc
			};

			this.props.addItemToOrder(item);
		}
		else {
			this.props.removeItemFromOrderByIndex(indexInOrder);
		}
	}

	getIndexInOrder() {
		const {_id} = this.props;
		const {orderedItems} = this.props;

		let indexInOrder = null;

		orderedItems.find((orderedItem, orderedItemIndex) => {
			if (_id === orderedItem._id) {
				indexInOrder = orderedItemIndex;
			}
		});

		return indexInOrder;
	}

	checkThatItemIsOrdered() {
		return (this.getIndexInOrder() !== null);
	}

	render() {
		const {name, price, rarity, type, imageSrc} = this.props;

		return (
			<div
				className={'item-search-card item-search-card--raised'}
				onClick={this.switchOrder}
			>
				<div className={'item-search-card--top-part'}>
					<div className={'top-part--visual-box'}>
						<ItemCoin
							imageSrc={imageSrc}
							rarity={rarity}
						/>
						<div
							className={`visual-box--action-box ${(this.checkThatItemIsOrdered()) && 'visual-box--action-box--active'}`}
						>
							<div className={'material-icons action-box--sign'}>
								<span>clear</span>
							</div>
						</div>
					</div>
				</div>
				<div className={'item-search-card--bot-part'}>
					<div className={'bot-part--name'}>{name}</div>
					{
						(this.checkThatItemIsOrdered()) &&
							<span>ORDERED</span>
					}
					<div className={'bot-part--description'}>
						<div className={'description--type'}>{type}</div>
						<div className={'description--price'}>{price}</div>
					</div>
				</div>
			</div>
		);
	}
}

ItemSearchCard.propTypes = {
	_id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	price: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	orderedItems: PropTypes.array, //redux,
	addItemToOrder: PropTypes.func, //redux
	removeItemFromOrderByIndex: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items: orderedItems} = state.order;
	return {orderedItems};
};

const mapDispatchToProps = {addItemToOrder, removeItemFromOrderByIndex};

export default connect(mapStateToProps, mapDispatchToProps)(ItemSearchCard);