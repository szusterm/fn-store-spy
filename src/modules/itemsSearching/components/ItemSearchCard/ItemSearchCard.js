import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {addItemToOrder, removeItemFromOrderByIndex} from '../../../orderCreator/actions';

import ItemCoin from '../../../../components/ItemCoin';

import './styles.scss';

export class ItemSearchCard extends Component {
	constructor() {
		super();

		this.switchOrder = this.switchOrder.bind(this);
		this.isItemOrdered = this.isItemOrdered.bind(this);
	}

	switchOrder() {
		const {fnbrId, name, rarity, imageSrc} = this.props;

		const indexInOrder = this.getIndexInOrder();

		if (indexInOrder === null) {
			const item = {
				fnbrId,
				name,
				rarity,
				imageSrc
			};

			this.props.addItemToOrder(item);
		}
		else {
			this.props.removeItemFromOrderByIndex(indexInOrder);
		}
	}

	getIndexInOrder() {
		const {fnbrId} = this.props;
		const {orderedItems} = this.props;

		let indexInOrder = null;

		orderedItems.find((orderedItem, orderedItemIndex) => {
			if (fnbrId === orderedItem.fnbrId) {
				indexInOrder = orderedItemIndex;
			}
		});

		return indexInOrder;
	}

	isItemOrdered() {
		return (this.getIndexInOrder() !== null);
	}

	render() {
		const {name, price, rarity, imageSrc} = this.props;

		return (
			<div className={'col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3'}>
				<button
					className={'item-search-card item-search-card--raised'}
					onClick={this.switchOrder}
				>
					<div className={'item-search-card--top-part'}>
						<div className={'top-part--visual-box'}>
							<ItemCoin
								imageSrc={imageSrc}
								rarity={rarity}
								name={name}
							/>
							<div
								className={`visual-box--action-box ${(this.isItemOrdered()) && 'visual-box--action-box--active'}`}
							>
								<div className={'material-icons action-box--sign'}>
									<span>{(this.isItemOrdered()) ? 'done' : 'add'}</span>
								</div>
							</div>
						</div>
					</div>
					<div className={'item-search-card--bot-part'}>
						<div className={'bot-part--name'}>{name}</div>
						<div className={'bot-part--description'}>
							<img
								className={'description--vbucks-icon'}
								src={'../images/vbucks.png'}
							/>
							<span className={'description--price'}>{price}</span>
						</div>
					</div>
				</button>
			</div>
		);
	}
}

ItemSearchCard.propTypes = {
	fnbrId: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	price: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	orderedItems: PropTypes.array, //redux,
	addItemToOrder: PropTypes.func, //redux
	removeItemFromOrderByIndex: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {items: orderedItems} = state.orderCreator;
	return {orderedItems};
};

const mapDispatchToProps = {addItemToOrder, removeItemFromOrderByIndex};

export default connect(mapStateToProps, mapDispatchToProps)(ItemSearchCard);