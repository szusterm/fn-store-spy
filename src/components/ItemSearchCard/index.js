import ItemSearchCard from './ItemSearchCard';

export default ItemSearchCard;