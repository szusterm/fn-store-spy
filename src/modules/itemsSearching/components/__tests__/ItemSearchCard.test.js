import React from 'react';
import {shallow} from 'enzyme';

import ItemCoin from '../../../../components/ItemCoin';

import {ItemSearchCard} from '../ItemSearchCard/ItemSearchCard';

const setup = (propOverrides) => {
	const props = Object.assign({
		_id: 'eg8er3478e',
		name: 'Super Axe',
		price: '2000',
		rarity: 'uncommon',
		type: 'axe',
		imageSrc: 'https://example.com',
		orderedItems: [], //redux
		addItemToOrder: () => true, //redux
		removeItemFromOrderByIndex: () => true //redux
	}, propOverrides);

	const wrapper = shallow(<ItemSearchCard {...props}/>);

	return {props, wrapper};
};

describe('ItemSearchCard Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('shows a name from prop', () => {
		const {wrapper, props} = setup({
			name: 'Normal Weapon'
		});

		const displayedName = wrapper.find('.bot-part--name').text();

		expect(displayedName).toBe(props.name);
	});

	it('shows a price from prop', () => {
		const {wrapper, props} = setup({
			price: '2000'
		});

		const displayedPrice = wrapper.find('.description--price').text();

		expect(displayedPrice).toBe(props.price);
	});

	it('shows a type from prop', () => {
		const {wrapper, props} = setup({
			type: 'axe'
		});

		const displayedType = wrapper.find('.description--type').text();

		expect(displayedType).toBe(props.type);
	});

	it('puts rarity prop to ItemCoin', () => {
		const {wrapper, props} = setup({
			rarity: 'uncommon'
		});

		const rarityInItemCoin = wrapper.find(ItemCoin).props().rarity;

		expect(rarityInItemCoin).toBe(props.rarity);
	});

	it('puts imageSrc to ItemCoin', () => {
		const {wrapper, props} = setup({
			imageSrc: 'https://example.com'
		});

		const imageSrcInItemCoin = wrapper.find(ItemCoin).props().imageSrc;

		expect(imageSrcInItemCoin).toBe(props.imageSrc);
	});

	it('after click the component, adds the item to an order, if it is not ordered', () => {
		const {wrapper, props} = setup({
			_id: 'abc45g',
			addItemToOrder: jest.fn(),
			orderedItems: []
		});

		wrapper.find('button').simulate('click');

		expect(props.addItemToOrder).toHaveBeenCalledTimes(1);
	});

	it('after click the component, removes the item from an order, if it is ordered', () => {
		const {wrapper, props} = setup({
			_id: 'superid',
			removeItemFromOrderByIndex: jest.fn(),
			orderedItems: [
				{_id: 'xddd'},
				{_id: 'superid'}
			]
		});

		wrapper.find('button').simulate('click');

		expect(props.removeItemFromOrderByIndex).toHaveBeenCalledWith(1);
	});

	it('shows confirmation, if the item is ordered', () => {
		const {wrapper, props} = setup({
			_id: 'superid',
			orderedItems: []
		});

		const confirmationClass = '.visual-box--action-box--active';
		const iconElement = '.action-box--sign span';
		const confirmationIcon = 'done';


		const confirmationWithoutOrder = wrapper.find(confirmationClass);
		const iconWithoutOrder = wrapper.find(iconElement).text();

		expect(confirmationWithoutOrder.exists()).toEqual(false);
		expect(iconWithoutOrder).not.toBe(confirmationIcon);

		wrapper.setProps({
			orderedItems: [{_id: props._id}]
		});

		const confirmationWithOrder = wrapper.find(confirmationClass);
		const iconWithOrder = wrapper.find(iconElement).text();

		expect(confirmationWithOrder.exists()).toEqual(true);
		expect(iconWithOrder).toBe(confirmationIcon);
	});
});