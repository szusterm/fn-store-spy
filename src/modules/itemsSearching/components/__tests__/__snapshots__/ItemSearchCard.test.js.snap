// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSearchCard Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemSearchCard
    _id="eg8er3478e"
    addItemToOrder={[Function]}
    imageSrc="https://example.com"
    name="Super Axe"
    orderedItems={Array []}
    price="2000"
    rarity="uncommon"
    removeItemFromOrderByIndex={[Function]}
    type="axe"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="item-search-card item-search-card--raised"
        onClick={[Function]}
      >
        <div
          className="item-search-card--top-part"
        >
          <div
            className="top-part--visual-box"
          >
            <ItemCoin
              imageSrc="https://example.com"
              rarity="uncommon"
            />
            <div
              className="visual-box--action-box false"
            >
              <div
                className="material-icons action-box--sign"
              >
                <span>
                  add
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="item-search-card--bot-part"
        >
          <div
            className="bot-part--name"
          >
            Super Axe
          </div>
          <div
            className="bot-part--description"
          >
            <div
              className="description--type"
            >
              axe
            </div>
            <div
              className="description--price"
            >
              2000
            </div>
          </div>
        </div>
      </button>,
      "className": "col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item-search-card--top-part"
          >
            <div
              className="top-part--visual-box"
            >
              <ItemCoin
                imageSrc="https://example.com"
                rarity="uncommon"
              />
              <div
                className="visual-box--action-box false"
              >
                <div
                  className="material-icons action-box--sign"
                >
                  <span>
                    add
                  </span>
                </div>
              </div>
            </div>
          </div>,
          <div
            className="item-search-card--bot-part"
          >
            <div
              className="bot-part--name"
            >
              Super Axe
            </div>
            <div
              className="bot-part--description"
            >
              <div
                className="description--type"
              >
                axe
              </div>
              <div
                className="description--price"
              >
                2000
              </div>
            </div>
          </div>,
        ],
        "className": "item-search-card item-search-card--raised",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="top-part--visual-box"
            >
              <ItemCoin
                imageSrc="https://example.com"
                rarity="uncommon"
              />
              <div
                className="visual-box--action-box false"
              >
                <div
                  className="material-icons action-box--sign"
                >
                  <span>
                    add
                  </span>
                </div>
              </div>
            </div>,
            "className": "item-search-card--top-part",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ItemCoin
                  imageSrc="https://example.com"
                  rarity="uncommon"
                />,
                <div
                  className="visual-box--action-box false"
                >
                  <div
                    className="material-icons action-box--sign"
                  >
                    <span>
                      add
                    </span>
                  </div>
                </div>,
              ],
              "className": "top-part--visual-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "imageSrc": "https://example.com",
                  "rarity": "uncommon",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="material-icons action-box--sign"
                  >
                    <span>
                      add
                    </span>
                  </div>,
                  "className": "visual-box--action-box false",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                      add
                    </span>,
                    "className": "material-icons action-box--sign",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "add",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": "span",
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="bot-part--name"
              >
                Super Axe
              </div>,
              <div
                className="bot-part--description"
              >
                <div
                  className="description--type"
                >
                  axe
                </div>
                <div
                  className="description--price"
                >
                  2000
                </div>
              </div>,
            ],
            "className": "item-search-card--bot-part",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Super Axe",
                "className": "bot-part--name",
              },
              "ref": null,
              "rendered": "Super Axe",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="description--type"
                  >
                    axe
                  </div>,
                  <div
                    className="description--price"
                  >
                    2000
                  </div>,
                ],
                "className": "bot-part--description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "axe",
                    "className": "description--type",
                  },
                  "ref": null,
                  "rendered": "axe",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2000",
                    "className": "description--price",
                  },
                  "ref": null,
                  "rendered": "2000",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="item-search-card item-search-card--raised"
          onClick={[Function]}
        >
          <div
            className="item-search-card--top-part"
          >
            <div
              className="top-part--visual-box"
            >
              <ItemCoin
                imageSrc="https://example.com"
                rarity="uncommon"
              />
              <div
                className="visual-box--action-box false"
              >
                <div
                  className="material-icons action-box--sign"
                >
                  <span>
                    add
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="item-search-card--bot-part"
          >
            <div
              className="bot-part--name"
            >
              Super Axe
            </div>
            <div
              className="bot-part--description"
            >
              <div
                className="description--type"
              >
                axe
              </div>
              <div
                className="description--price"
              >
                2000
              </div>
            </div>
          </div>
        </button>,
        "className": "col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="item-search-card--top-part"
            >
              <div
                className="top-part--visual-box"
              >
                <ItemCoin
                  imageSrc="https://example.com"
                  rarity="uncommon"
                />
                <div
                  className="visual-box--action-box false"
                >
                  <div
                    className="material-icons action-box--sign"
                  >
                    <span>
                      add
                    </span>
                  </div>
                </div>
              </div>
            </div>,
            <div
              className="item-search-card--bot-part"
            >
              <div
                className="bot-part--name"
              >
                Super Axe
              </div>
              <div
                className="bot-part--description"
              >
                <div
                  className="description--type"
                >
                  axe
                </div>
                <div
                  className="description--price"
                >
                  2000
                </div>
              </div>
            </div>,
          ],
          "className": "item-search-card item-search-card--raised",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="top-part--visual-box"
              >
                <ItemCoin
                  imageSrc="https://example.com"
                  rarity="uncommon"
                />
                <div
                  className="visual-box--action-box false"
                >
                  <div
                    className="material-icons action-box--sign"
                  >
                    <span>
                      add
                    </span>
                  </div>
                </div>
              </div>,
              "className": "item-search-card--top-part",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ItemCoin
                    imageSrc="https://example.com"
                    rarity="uncommon"
                  />,
                  <div
                    className="visual-box--action-box false"
                  >
                    <div
                      className="material-icons action-box--sign"
                    >
                      <span>
                        add
                      </span>
                    </div>
                  </div>,
                ],
                "className": "top-part--visual-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "imageSrc": "https://example.com",
                    "rarity": "uncommon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="material-icons action-box--sign"
                    >
                      <span>
                        add
                      </span>
                    </div>,
                    "className": "visual-box--action-box false",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <span>
                        add
                      </span>,
                      "className": "material-icons action-box--sign",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "add",
                      },
                      "ref": null,
                      "rendered": "add",
                      "type": "span",
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="bot-part--name"
                >
                  Super Axe
                </div>,
                <div
                  className="bot-part--description"
                >
                  <div
                    className="description--type"
                  >
                    axe
                  </div>
                  <div
                    className="description--price"
                  >
                    2000
                  </div>
                </div>,
              ],
              "className": "item-search-card--bot-part",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Super Axe",
                  "className": "bot-part--name",
                },
                "ref": null,
                "rendered": "Super Axe",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="description--type"
                    >
                      axe
                    </div>,
                    <div
                      className="description--price"
                    >
                      2000
                    </div>,
                  ],
                  "className": "bot-part--description",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "axe",
                      "className": "description--type",
                    },
                    "ref": null,
                    "rendered": "axe",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2000",
                      "className": "description--price",
                    },
                    "ref": null,
                    "rendered": "2000",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
